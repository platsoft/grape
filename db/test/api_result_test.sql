
\echo 'Test 1'
-- {"status":"OK"}
SELECT grape.api_success();

\echo 'Test 2'
-- {"status":"ERROR","message":"An error","code":-10}
SELECT grape.api_result_error('An error', -10);

\echo 'Test 3'
-- {"status":"OK","data":{"something":{"aaa":10}}}
SELECT grape.api_success('data', '{"something":{"aaa":10}}'::JSON);

\echo 'Test 4'
SELECT grape.api_result((true, 'Meh', '{"something":{"aaa":10}}'::JSON)::grape.grape_result_type);

\echo 'Test 5'
SELECT grape.api_result((false, 'Meh', '{"code":-10}'::JSON)::grape.grape_result_type);

\echo 'Test 6'
SELECT grape.api_result((false, 'Meh', NULL::JSON)::grape.grape_result_type);

\echo 'Test 7'
SELECT grape.api_success('{"something_numeric":30,"something_text":"bah"}'::JSON);

\echo 'Test 8'
SELECT grape.api_result_error('An error', -20);

