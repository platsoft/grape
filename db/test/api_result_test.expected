Test 1
{"status":"OK"}
Test 2
{"status":"ERROR","message":"An error","code":-10}
Test 3
{"status":"OK","data":{"something":{"aaa":10}}}
Test 4
{"status":"OK","data":{"something":{"aaa":10}}}
Test 5
{"status":"ERROR","message":"Meh","code":-10}
Test 6
{"status":"ERROR","message":"Meh","code":-1}
Test 7
{"status": "OK", "something_text": "bah", "something_numeric": 30}
Test 8
{"status":"ERROR","message":"An error","code":-20}
